tools/karl.cpp

```diff
@@ -294,10 +300,25 @@ bool capnp_import_dirs_flag = false;
 // handle command line arguments
 void handle_arguments (int argc, char ** argv)
 {
+  // retreive user home dir in a crossplatform way
+  std::string full_path = madara::utility::expand_envs("$(HOME)/.karl/karl.cfg");
```

Comment:
This global should have a more descriptive name. `full_path` could be anything. Also, it's aliased by an argument of `load_config_file`, which is confusing. Might as well be `const` as well.

Reply:
Done...